index.js

```diff
@@ -209,6 +209,47 @@ function filePathFactory(file) {
 }
 
 /**
+ * If the vFile has a extension, it will be prefixed with the filename,
+ * if applicable. If the vFile does not have extension,
+ * only `filename` will be returned.
+ * Otherwise, an empty string is returned.
+ *
+ * @private
+ * @param {VFile} file - Virtual file.
+ * @return {Function} - `fileName` getter.
+ */
+function fileBasenameFactory(file) {
+    /**
+     * Get the filename with extantion.
+     *
+     * @example
+     *   var file = new VFile({
+     *     'filename': 'example',
+     *     'extension': 'txt'
+     *   });
+     *
+     *   String(file.fileName); // example.txt
+     *   file.fileName() // example.txt
+     *
+     * @memberof {VFile}
+     * @property {Function} toString - Itself.
+     * @return {string} - name of file with extantion.
+     */
+    function fileBasename() {
+        if (file.filename || file.extension) {
+            return file.filename +
+                (file.extension ? '.' + file.extension : '');
+        }
+
+        return '';
+    }
+
+    fileBasename.toString = fileBasename;
+
```

Comment:
There’s no need to wrap this in a factory here, you can just expose a function `basename` on `VFile#`.

The fact that `filePath` is wrapped is to support ESLint reporters, but they don’t have a `basename` so that’s superfluous.